src/toil/batchSystems/kubernetes.py

```diff
@@ -0,0 +1,668 @@
+                    
+            # Remember the continuation token, if any
+            token = getattr(results.metadata, 'continue', None)
+
+            if token is None:
+                # There isn't one. We got everything.
+                break
+                
+    def _getPodForJob(self, jobObject):
+        """
+        Get the pod that belongs to the given job, or None if the job's pod is
+        missing. The pod knows about things like the job's exit code.
+
+        :param kubernetes.client.V1Job jobObject: a Kubernetes job to look up
+                                       pods for.
+
+        :return: The pod for the job, or None if no pod is found.
+        :rtype: kubernetes.client.V1Pod
+        """
+        
+        token = None
+        
+        # Work out what the return code was (which we need to get from the
+        # pods) We get the associated pods by querying on the label selector
+        # `job-name=JOBNAME`
+        query = 'job-name={}'.format(jobObject.metadata.name)
+        
+        while True:
+            # We can't just pass e.g. a None continue token when there isn't
+            # one, because the Kubernetes module reads its kwargs dict and
+            # cares about presence/absence. So we build a dict to send.
+            kwargs = {'label_selector': query}
+            if token is not None:
+                kwargs['_continue'] = token
+            results = self.coreApi.list_namespaced_pod(self.namespace, **kwargs)
+            
+            for pod in results.items:
+                # Return the first pod we find
+                return pod
+                    
+            # Remember the continuation token, if any
+            token = getattr(results.metadata, 'continue', None)
+        
+            if token is None:
+                # There isn't one. We got everything.
+                break
+                
+        # If we get here, no pages had any pods.
+        return None
+
+    def _getLogForPod(self, podObject):
+        """
+        Get the log for a pod.
+
+        :param kubernetes.client.V1Pod podObject: a Kubernetes pod with one
+                                       container to get the log from.
+
+        :return: The log for the only container in the pod.
+        :rtype: str
+
+        """
+
+        return self.coreApi.read_namespaced_pod_log(podObject.metadata.name,
+                                                    namespace=self.namespace)
+   
+    def _getIDForOurJob(self, jobObject):
+        """
+        Get the JobID number that belongs to the given job that we own.
+
+        :param kubernetes.client.V1Job jobObject: a Kubernetes job object that is a job we issued.
+
+        :return: The JobID for the job.
+        :rtype: int
+        """
+
+        return int(jobObject.metadata.name[len(self.jobPrefix):])
+    
+
+    def _genPods(self):
+        for j in _ourJobObjects:
+            yield self._getPodForJob(j)
+
+    def getUpdatedBatchJob(self, maxWait):
```

Comment:
Nit: It might be worth breaking this up.

Reply:
I think we can address this when we hopefully swap over to a watch-based implementation.